android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java

```diff
@@ -311,8 +314,31 @@ public void init() {
                 }
             }
 
-            getAllNetworks(mJsonRpcService, mSharedData.getSupportedCryptoCoins(),
-                    allNetworks -> { _mCryptoNetworks.postValue(new ArrayList<>(allNetworks)); });
+            getAllNetworks(
+                    mJsonRpcService, mSharedData.getSupportedCryptoCoins(), cryptoNetworks -> {
+                        _mCryptoNetworks.postValue(cryptoNetworks);
+
+                        List<NetworkInfo> primary = new ArrayList<>();
+                        List<NetworkInfo> secondary = new ArrayList<>();
+                        List<NetworkInfo> test = new ArrayList<>();
+                        NetworkLists networkLists =
+                                new NetworkLists(cryptoNetworks, primary, secondary, test);
+                        for (NetworkInfo networkInfo : cryptoNetworks) {
+                            if (WalletConstants.SUPPORTED_TOP_LEVEL_CHAIN_IDS.contains(
+                                        networkInfo.chainId)) {
+                                primary.add(networkInfo);
+                            } else if (WalletConstants.KNOWN_TEST_CHAIN_IDS.contains(
+                                               networkInfo.chainId)) {
+                                test.add(networkInfo);
+                            } else if (!WalletConstants.SUPPORTED_TOP_LEVEL_CHAIN_IDS.contains(
```

Comment:
we can just do `else { ...` without conditions check as they always pass on that stage because of 2 conditions above.

Reply:
It's just a matter of taste, but I would probably add a forth `else` where we log an error (as it should never happen), or throw an exception.

Reply:
replaced with `else` https://github.com/brave/brave-core/pull/19467/commits/d194853b1cbba8f24d94fd581ef25b76602d3727